jenkins/data-prepper/distribution-artifacts.jenkinsfile

```diff
@@ -0,0 +1,80 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'BRANCH', description: 'Can be: <branchName>, <commitId>, refs/tags/<tagName>', trim: true)
+    }
+    stages {
+        stage('Build Data Prepper') {
+            steps {
+                script {
+                    checkout changelog: false, poll: false, scm:
+                            [$class: 'GitSCM', branches: [[name: BRANCH]], userRemoteConfigs: [[url: 'https://github.com/opensearch-project/data-prepper.git']]]
+                    sh './gradlew clean :release:archives:linux:linuxTar -Prelease'
+                    artifactsPath = "$WORKSPACE/release/archives/linux/build/distributions"
+                    artifactFileNames = sh(script: "find ${artifactsPath} | sed -n \"s|^${artifactsPath}/||p\"", returnStdout: true).split()
+                }
+            }
+        }
+        stage('Sign Artifacts') {
+            steps {
+                script {
+                    signatureType = '.sig'
+                    signArtifacts(
+                            artifactPath: "${artifactsPath}",
+                            signatureType: signatureType,
+                            distributionPlatform: 'linux'
+                    )
+                }
+            }
+        }
+        stage('Upload Artifacts to Staging Distribution Bucket') {
+            steps {
+                script {
+                    uploadPath = ([
+                            "${JOB_NAME}",
+                            "${VERSION}",
+                            "${BUILD_NUMBER}",
+                            "builds",
+                            "signed"
```

Comment:
Do we need `signed` in the upload path? Do we have any unsigned artifacts in `builds` folder

Reply:
I believe @abhinavGupta16 suggested this path structure. I assume that we probably want to allow room for other artifacts, and using a directory structure can help with that.

Reply:
But, I am flexible in regard to the structure.

Reply:
I had the `signed` folder in the artifact path because the artifacts for windows and mac end up with the same filename unlike linux with has a separate file. So this wouldn’t need any change as and when it's extended

Reply:
Okay! I thought all signed artifacts were generated and stored in local and then signed and uploaded to `builds` folder. Anything that makes it less confusing works. Thanks!